### PR TITLE
This makes a single coordinate __repr__ like that of arrays of coordinates

### PR DESCRIPTION
**Pull request by adrn**
This is a fix based on #3298 -- I had to add some subtle numpy `array2string()` hacks, so we may want to keep an eye on this if numpy/numpy#5463 gets replies.

The new behavior is:

    >>> coord.SkyCoord(ra=13.5135*u.hourangle, dec=-10.4124*u.degree)
    <SkyCoord (ICRS): (ra, dec) in deg
        (202.7025, -10.4124)>

### CODE REVIEW

**Comment by adrn** [comment]
cc @eteq @mhvk

File: astropy/coordinates/baseframe.py [hunk 1]
```diff
@@ -810,11 +810,13 @@ def represent_as(self, new_representation, in_frame_units=False):
         >>> from astropy.coordinates import SkyCoord, CartesianRepresentation
         >>> coord = SkyCoord(0*u.deg, 0*u.deg)
         >>> coord.represent_as(CartesianRepresentation)
-        <CartesianRepresentation x=1.0, y=0.0, z=0.0>
+        <CartesianRepresentation (x, y, z) in
```
Comment (mhvk) [inline astropy/coordinates/baseframe.py]: Not really a problem with this PR, but strange not to have a unit here -- either add "dimensionless units" or maybe just remove "in"?

File: astropy/coordinates/representation.py [hunk 2]
```diff
@@ -153,29 +153,44 @@ def _unitstr(self):
...
+                                            ")"}
+
+        arrstr = np.array2string(v, formatter=fmt,
+                                 separator=', ',
+                                 prefix=prefixstr)
+
+        if self._values.shape == ():
+            arrstr = arrstr[1:-1]
+
+        return '<{0} ({1}) in {2:s}\n{3}{4}>'.format(
```
Comment (mhvk) [inline astropy/coordinates/representation.py]: I guess before this line I would do either one of the following:
``
unitstr = "in " + self._unitstr if self._unitstr is not None else ""
```
or
```
unitstr = 'dimensionles' if self._unitstr is None else self._unitstr
```

Of course, in the first example, the format string should be adjusted appropriately.

File: astropy/coordinates/baseframe.py (hunk 1 above)
Comment (adrn) [inline astropy/coordinates/baseframe.py]: Yea, I'm in favor of just dropping the "in" -- I could tack that on to this PR? @eteq ?

**Comment by adrn** [comment]
It looks like the tests are failing in Python 3 and in Python 2 with Numpy 1.6 for different reasons. The Numpy 1.6 failure is because the `formatter` kwarg in `array2string()` was added in 1.7, so I need to figure out a workaround for that. Not sure what the failure in Python 3 is, maybe a difference in how floats are printed?

**Comment by eteq** [comment]
@adrn - see #2271 where I ran into that `formatter` problem and I think what I did there was just have crappier behavior for 1.6...

**Comment by adrn** [comment]
@eteq Ok, I'll take a similar approach here? Should I just short-circuit the effected tests if Numpy version < 1.7?

**Comment by eteq** [comment]
yeah, probably that makes sense

**Comment by adrn** [comment]
@eteq hmm, the other failure is because in Python 3:

    >>> str(2.00000000000003)
    '2.00000000000003'

versus Python 2:

    >>> str(2.00000000000003)
    '2.0'

Any thoughts there? Two possibilities in my mind: 

1) use a precision of our own, e.g., instead of doing str(x) do "{:.8f}".format(x) and strip trailing zeros
2) back off from being fancy and just suck up the machine precision hanging digits -- this is uglier, but requires a lot less hacking

**Comment by adrn** [comment]
@eteq whoops, it seems like they just have different tolerances for when to truncate:

__Python 3.4__

    >>> str(2 + 1E-15)
    '2.000000000000001'
    >>> str(2 + 1E-16)
    '2.0'

__Python 2.7.8__

    >>> str(2 + 1E-11)
    '2.00000000001'

    >>> str(2 + 1E-12)
    '2.0'

**Comment by eteq** [comment]
@adrn - yeah, I think it's ok to just accept that it will look uglier for numpy 2.6 - make sure to still test it doesn't actually like completely error out or something, but otherwise just be lax about how well it matches there.

But there's also a bunch of errors in python 2.6 ?  Not sure what that's about...

**Comment by adrn** [comment]
@eteq ha, supporting old versions is fun! Here's the Python 2.6 failure:

__Python 2.6.9__

    >>> "fasdf{}".format("")
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    ValueError: zero length field name in format

__Python 2.7.8__

    >>> "fasdf{}".format("")
    'fasdf'

**Comment by astrofrog** [comment]
@adrn - in Python 2.6 you cannot write ``{}``, you always need to explicitly specify the field, e.g. ``{0}`` in your case.

**Comment by adrn** [comment]
@eteq I think this is ready for you to take a look at!

File: astropy/coordinates/baseframe.py (hunk 1 above)
Comment (eteq) [inline astropy/coordinates/baseframe.py]: I agree dropping the "in" makes sense - feel free to do that here or separately

**Comment by eteq** [comment]
@adrn - aside from removing the "in" for dimensionless units, I think this is good to go (and the "in" part can be addressed separately if you prefer).

@astrofrog - do we want to sneak this in for 1.0?

**Comment by adrn** [comment]
@eteq I can remove the `in` but probably better to do in a separate PR -- don't want this one to stagnate and I probably won't get around to it for a few days...

**Comment by eteq** [comment]
Alright, it's passing and this has been even before the last RC so I'll make a command decision to include it.

**Pull request merged**